docs/openapi-qtsp.yaml

```diff
@@ -0,0 +1,185 @@
+openapi: 3.0.3
+
+info:
+  title: Firma con IO QTSP
+  description: Any APIs called by IO Backend to QTSP
+  version: 1.0.0
+
+paths:
+
+  /tos:
+    summary: Request terms of service
+    get:
+      operationId: GetTos
+      tags:
+        - QTSP
+
+      responses:
+        '200':
+          description: Terms of service to be accepted
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/TOS"
+        
+        default:
+          description: Generic error
+      
+      
+      
+  /login:
+    summary: Request login to QTSP
+    post:
+      operationId: GetChallenge
+      tags:
+        - QTSP
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Auth'
+      responses:
+        '200':
+          description: Challenge generated
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Challenge"
+        400:
+          description: Bad request
+        401:
+          description: SAML assertion not verified
+        404:
+          description: User not found
+        default:
+          description: Generic error
+      
+  /sign:
+    summary: Send signature request
+    post:
+      operationId: GetSignature
+      tags:
+        - QTSP
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/SignatureRequest'
+      responses:
+        '200':
+          description: Challenge generated
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/SignatureResponse"
+        400:
+          description: Bad request
+        401:
+          description: Challenge sign invalid
+        404:
+          description: Challenge not found
+        default:
+          description: Generic error
+
+        
+          
+components:
+  schemas:
+  
+    TOS:
+      type: object
+      required:
+        - tos
+      properties:
+        tos:
+          type: array
+          items: 
+            type: string
+          description: "List of terms of service"
+        privacyLink:
+          type: string
+          format: uri
+          description: "Link to privacy policy"
+
+          
+    Auth:
+      type: object
+      required:
+        - SAMLAssertion
+        - publicKey
+      properties:
+        SAMLAssertion:
+          type: string
+          format: byte
+          description: "SAML assertion in base64"
```

Comment:
To request the challenge we must necessarily send SAMLAssertion and publicKey. The data relating to the fiscal code and the hashes to be signed will be sent together with the signed challenge. If the QTSP still has to generate the challenge, it makes no sense to send the file hashes.

Reply:
Why must we "necessarily send SAMLAssertion and publicKey" ? I'm suggesting to send the hashes since in my view
the challenge is related to the tuple (user -> files to be signed).